tests/test_association_task.py

```diff
@@ -313,22 +323,13 @@ def test_score_and_match(self):
         # except the last DIAObject in this collection which should be
         # newly created during the matching step and contain only one
         # DIASource.
-        match_result = dia_collection.match(dia_sources, score_struct)
-        updated_ids = match_result.updated_and_new_dia_object_ids
-        self.assertEqual(len(dia_collection.dia_objects), 6)
+        match_result = assoc_task.match(dia_objects, dia_sources, score_struct)
+        updated_ids = match_result.associated_dia_object_ids
+        self.assertEqual(len(updated_ids), 5)
         self.assertEqual(match_result.n_updated_dia_objects, 4)
         self.assertEqual(match_result.n_new_dia_objects, 1)
         self.assertEqual(match_result.n_unassociated_dia_objects, 1)
 
-        # We created a new DIAObject in the collection hence the last
-        # DIAObject in this collection is new and contains only one
-        # DIASource. All others contain two.
-        self.assertEqual(
-            dia_collection.get_dia_object(updated_ids[-1]).n_dia_sources, 1)
-        for obj_id in updated_ids[0:-1]:
-            self.assertEqual(
-                dia_collection.get_dia_object(obj_id).n_dia_sources, 2)
-
```

Comment:
Why don't you need these assertions anymore?

Reply:
In the previous version, the match method created new DIAObjects and associated the DIASources into the existing DIAObjects. This happens in the update_dia_objects method instead now so these assertions don't make sense here anymore.